projects/igniteui-angular/src/lib/services/exporter-common/base-export-service.ts

```diff
@@ -344,35 +356,72 @@ export abstract class IgxBaseExporter {
         });
     }
 
+    private calculateColumnSpans(column: IColumnInfo, mapRecord: IColumnList, span: number) {
+        if (column.headerType === HeaderType.MultiColumnHeader && column.skip) {
+            const columnGroupChildren = mapRecord.columns.filter(c => c.columnGroupParent === column.columnGroup);
+
+            columnGroupChildren.forEach(cgc => {
+                if (cgc.headerType === HeaderType.MultiColumnHeader) {
+                    cgc.columnSpan = 0;
+                    cgc.columnGroupParent = null;
+                    cgc.skip = true;
+
+                    this.calculateColumnSpans(cgc, mapRecord, cgc.columnSpan);
+                } else {
+                    cgc.skip = true;
+                }
+            });
+        }
+
+        const targetCol = mapRecord.columns.filter(c => column.columnGroupParent !== null && c.columnGroup === column.columnGroupParent)[0];
+        if (targetCol !== undefined) {
+            targetCol.columnSpan -= span;
+
+            if (targetCol.columnGroupParent !== null) {
+                this.calculateColumnSpans(targetCol, mapRecord, span);
+            }
+
+            if (targetCol.columnSpan === 0) {
+                targetCol.skip = true;
+            }
+        }
+    }
+
```

Comment:
The behavior seems OK now. I suggest a little optimization of the code as follows:
```suggestion
    /** Update column spans of parent columns & skip child columns exporting. */
    private calculateColumnSpans(column: IColumnInfo, mapRecord: IColumnList, span: number) {
        if (column.columnGroupParent !== null) {
            const parentCol = mapRecord.columns.filter(c => c.columnGroup === column.columnGroupParent)[0];
            this.calculateColumnSpans(parentCol, mapRecord, span);
        }

        if (column.columnGroup && column.skip) {
            this.skipAllChildColumns(column, mapRecord);
        }

        column.columnSpan -= span;
    }

    /** Recursively skip all child columns of a group column. */
    private skipAllChildColumns(column: IColumnInfo, mapRecord: IColumnList) {
        const children = mapRecord.columns.filter(c => c.columnGroupParent === column.columnGroup);
        children.forEach(c => {
            c.skip = true;
            if (c.columnGroup) this.skipAllChildColumns(c, mapRecord);
        });
    }
```